src/main/java/org/commcare/cases/util/StorageBackedTreeRoot.java

```diff
@@ -134,7 +135,7 @@ private void collectNativePredicateProfiles(Vector<XPathExpression> predicates,
         for (XPathExpression xpe : predicates) {
             //what we want here is a static evaluation of the expression to see if it consists of evaluating
             //something we index with something static.
-            if (xpe instanceof XPathEqExpr && ((XPathEqExpr)xpe).op == XPathEqExpr.EQ) {
+            if (xpe instanceof XPathEqExpr) {
```

Comment:
One thing that might also be worth evaluating is the pattern

`not(a=b)`

I don't remember if the `not` function is parsed as a straight-up xpath function or whether it's parsed as syntactic sugar, but if it's the former it should be fairly straightforward to just check `xpe instanceof XPathNotFunctexpr` or whatnot, and then check the single argument as being a `XPathEqExpr` with an EQ op like we do here.